mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDao.java

```diff
@@ -287,6 +287,14 @@
    */
   Future<Boolean> deleteRecordsBySnapshotId(String snapshotId, String tenantId);
 
+  /**
+   *  Performs purge the 'DELETED' records, and its related 'OLD' records
+   *
+   * @param tenantId tenant id
+   * @return future
+   */
+  Future<Void> cleanRecords(String tenantId);
```

Comment:
It's needed to keep soft-deleted records for some time. In the approach described by SA, he suggests removing records that are older than 7 days. Also, he suggests limiting the maximum amount of records deleted for example by 500 records in a time.